### PR TITLE
[6610] Deprecated RBUDP (4-3-stable)

### PR DESCRIPTION
**Pull request by JustinKyleJames**
I am opening this so that everyone can take a look.  I am not sure if I did this correctly or not.  When someone calls the various -Q options via icommands (iput, iget, icp, irepl), they will get a warning message that states the following:

```
Warning: RBUDP (-Q) is deprecated. Its use should be avoided.
```
The following will show up in the server log:

```
Warning: RBUDP transfer option was called.  This has been deprecated.
```
I am not sure I implemented the client side the correct way.  Furthermore, I am not sure that this code has any effect.

```
    const irods::at_scope_exit add_deprecated_api_warning_message{[&] {
        addRErrorMsg(&rsComm->rError, DEPRECATED_API, "RBUDP has been deprecated.");
    }};
```

I was hoping this would make it back to the client so that this warning could be seen by non-icommands.

Finally, there are many methods that are called by the ones deprecated here that could also be deprecated.  Not sure how far I should go with that.

### CODE REVIEW

File: server/core/src/miscServerFunct.cpp
Comment (alanking) [inline server/core/src/miscServerFunct.cpp:2045]: I'm not sure that we need to log a message for this. The rError message should be enough.

**Review by alanking — COMMENTED** [review]
I think that the rError message you implemented is fine. What makes you think it won't make it back to the client? Since the `RcComm` is part of communicating with the server, I *think* most clients will have some form of `rError` stack and the message will be in there, even for non-icommands.

File: server/core/src/miscServerFunct.cpp
Comment (alanking) [reply server/core/src/miscServerFunct.cpp:2045]: That being said, I'm not opposed to it. I defer to others.

**Comment by JustinKyleJames** [comment]
> I think that the rError message you implemented is fine. What makes you think it won't make it back to the client? Since the `RcComm` is part of communicating with the server, I _think_ most clients will have some form of `rError` stack and the message will be in there, even for non-icommands.

I am not seeing that logged when I run icommands with the -Q option.  That is why I implemented specific warning log messages in the parse command line methods.

**Comment by alanking** [comment]
Oh, I see. And you do see the warning message in the log? Hmm... I wonder if the message isn't making it back because it establishes a separate connection with the client and the rError stack isn't copied back to the original 1247 connection.

Perhaps we should just print a message in the iCommands which use the option. I'm not sure whether any other clients have RBUDP implemented, but given that traditional parallel transfer is implemented for very few, if any, I would guess that RBUDP is not either.

Thoughts?

**Comment by JustinKyleJames** [comment]
> Perhaps we should just print a message in the iCommands which use the option. I'm not sure whether any other clients have RBUDP implemented, but given that traditional parallel transfer is implemented for very few, if any, I would guess that RBUDP is not either.

I have that message printed in the iCommands so that should be covered.

File: lib/core/src/irods_parse_command_line_options.cpp
Comment (korydraughn) [inline lib/core/src/irods_parse_command_line_options.cpp:154]: Let's remove this since the help text states the option is deprecated.

File: lib/core/src/parseCommandLine.cpp
Comment (korydraughn) [inline lib/core/src/parseCommandLine.cpp:421]: Let's remove this since the help text states the option is deprecated.

**Comment by alanking** [comment]
> I have that message printed in the iCommands so that should be covered.

Oh, I see it now. But you said that you're not seeing the messages when the options are used with iCommands. Are you only referring to the rError message from the API?

**Review by korydraughn — COMMENTED** [review]
Nothing strange caught my eye.

File: lib/core/src/irods_parse_command_line_options.cpp
Comment (alanking) [reply lib/core/src/irods_parse_command_line_options.cpp:154]: The rules have changed: https://github.com/irods/irods_docs/issues/124#issuecomment-2023055718

Comment (JustinKyleJames) [reply lib/core/src/irods_parse_command_line_options.cpp]: I removed all warning messages.

File: lib/core/src/parseCommandLine.cpp
Comment (JustinKyleJames) [reply lib/core/src/parseCommandLine.cpp]: done

File: server/core/src/miscServerFunct.cpp
Comment (JustinKyleJames) [reply server/core/src/miscServerFunct.cpp]: Done

File: server/core/src/miscServerFunct.cpp
Comment (JustinKyleJames) [reply server/core/src/miscServerFunct.cpp]: It wasn't necessary.  I removed these pragmas.

File: server/core/src/miscServerFunct.cpp
Comment (JustinKyleJames) [reply server/core/src/miscServerFunct.cpp]: It wasn't necessary.  I removed these pragmas.

**Review by alanking — COMMENTED** [review]
Seems good to me. Please squash to taste

**Comment by korydraughn** [comment]
Please run clang-format on the changes.

**Comment by korydraughn** [comment]
So there's no `rc/rs` APIs for RBUDP? It's fine if the answer is yes.

**Comment by alanking** [comment]
Well... I'm not sure how one deprecates a whole directory, but we do have lib/rbudp which has some... stuff... Not sure how far to go with it.

**Comment by korydraughn** [comment]
If that directory contains implementation details then we don't need to touch those. We should focus on the public items.

If there aren't any rc/rs APIs then we're ready to go. And considering the release notes will contain a statement about deprecating RBUDP, we're good.

**Comment by korydraughn** [comment]
Please update the commit message to say `Deprecated RBUDP` so it covers all RBUDP related things.

**Comment by JustinKyleJames** [comment]
> Please update the commit message to say `Deprecated RBUDP` so it covers all RBUDP related things.

Done

**Comment by JustinKyleJames** [comment]
> If that directory contains implementation details then we don't need to touch those. We should focus on the public items.
> 
> If there aren't any rc/rs APIs then we're ready to go. And considering the release notes will contain a statement about deprecating RBUDP, we're good.

I could not find any rc/rs API's for it.

**Review by alanking — APPROVED** [review]
Seems good to me. Please await one more approval before #'ing.

**Review by korydraughn — APPROVED** [review]
Pound it!

**Comment by korydraughn** [comment]
Don't forget to run clang-format.

**Comment by JustinKyleJames** [comment]
> Pound it!

Done

**Comment by korydraughn** [comment]
Please cherry-pick to the main branch, no #'s

**Pull request merged**
